packages/client/src/components/dataset/RenameDatasetDialog.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { ReactElement } from "react";
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton, MenuItem,
+  TextField,
+  Tooltip
+} from "@material-ui/core";
+import gql from "graphql-tag";
+import { useApolloClient, useMutation } from "@apollo/client";
+
+const RENAME_DATASET_MUTATION = gql`
+    mutation RenameDataset($id: ID!, $newName: String!) {
+        renameDataset(id: $id, newName: $newName) {
+            id
+            name
+        }
+    }
+`;
+
+export default function RenameDatasetDialog(props: { id: string }): ReactElement {
+  const [open, setOpen] = React.useState(false);
+  const [newName, setNewName] = React.useState("");
+  const [renameDataset] = useMutation(RENAME_DATASET_MUTATION);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const handleRename = () => {
+    renameDataset({ variables: { id: props.id, newName } });
+    handleClose();
+  };
+
+  return (
+    <>
+      <MenuItem onClick={handleClickOpen}>
```

Comment:
I believe that since the dialog is wrapped inside the MenuItem, and the MenuItem sets open to true every time it is clicked, hitting cancel or rename will set open to true since you are technically clicking the menu. You could fix this by moving the </MenuItem> up so that the Dialog is not inside it.

Reply:
This fixed the issue, thank you!